src/main/webapp/app/shared/util/base64.util.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Decodes a base64url encoded string into an ArrayBuffer.
+ * Base64url is a URL-safe variant of base64 that uses '-' instead of '+' and '_' instead of '/'.
+ *
+ * @param base64url - The base64url encoded string to decode
+ * @returns The decoded data as ArrayBuffer
+ * @throws Error if the input is not a valid base64url string
+ */
+export function decodeBase64url(base64url: any): ArrayBuffer {
```

Comment:
I agree to code rabbit, that we should improve type safety and documentation in a follow-up PR

Reply:
Fixed in a follow-up. I also added client tests